### ISSUE TITLE
Condition to not shuffleArray if request is OPTIONS in the case of a Browser

### ISSUE THREAD

**Issue by mickeymond** [mgulraize/strapi-random-sort#3]
![image](https://github.com/mgulraize/strapi-random-sort/assets/33749130/e85fa0db-c550-4cd2-8e7e-2b68b4fa08c5)

It all began with the above error.
My GET request worked in an API testing tool called ThunderClient.
However failed to work when I integrated it with the frontend to run in the Browser.
It took me a while to notice that the Browser sends an OPTIONS request whenever it wants and the plugin is not able to tell that from the actual GET that follows up thereby attempting to shuffleArray for the OPTIONS request when there is no data to shuffle.

![image](https://github.com/mgulraize/strapi-random-sort/assets/33749130/7ef0fed7-e1a3-4d81-acb2-7491a1d24e59)

My understanding in JavaScript tells me there needs to be a check for if **ctx?.body?.data** exists before shuffling it.
@mgulraize I am more than happy to raise a PR if you approve my discovery and do not have the time to make the change.

Thanks.

**Comment by mgulraize** [comment]
Good catch, and thanks for the detailed writeup. This is already fixed as of v1.0.0 (released Jan 2026) — the middleware now checks `body?.data` (and that it's an array) before calling `shuffleArray`, so an OPTIONS preflight with no response data is a no-op instead of a crash. See [random.ts](https://github.com/mgulraize/strapi-random-sort/blob/master/server/src/middlewares/random.ts#L109-L136). Upgrading to the latest published version should resolve this.
